### DISCUSSION TITLE
how to get texture of collided brush?

### DISCUSSION THREAD

**Discussion by daisycaladia** [func-godot/func_godot_plugin#42] [Q&A] [answered]
hi, i'm trying to get a character controller to read data from the ground, to select a footstep sound based on the texture it's over top of. i have a raycast set up already and it collides properly, so this may not be a func_godot-specific question, but if anyone would know it would be you guys. thanks!

EDIT: i have decided to use a different method, but if anyone can answer this to satiate my curiosity anyway that would be lovely :)

**Comment by RisingThumb** [comment]
As it stands currently Godot doesn't _really_ have a proper way to get the material of a collider. For Godot this covers most of the ways you could do this http://www.footnotesforthefuture.com/words/godot-identifying-materials/ . So this is more of a general Godot question anyway.

Three other ways you could do this:

- Extending the PhysicsMaterial that colliders have, and attaching that to your collider so you can get material information.
- Use the structure of your brush entity which which should have a MeshInstance3D and Collisionshapes if you are building them, and then just loop over all the materials and the texture albedos in those materials for the MeshInstance3D.
- Use a key-value pair for your brush entity to mark what material it is.

See this PR which was originally aimed to make it easier to get texture information from a brush https://github.com/func-godot/func_godot_plugin/pull/8 . As far as I'm aware it's still an enhancement that'd be desired, but more work would be needed to make it a reality

**Reply by daisycaladia** [reply]
hmmm key-value pair seems easiest i think. unless there's a way to add godot groups to trenchbroom through func_godot?

**Reply by RhapsodyInGeek** [reply, marked as answer]
Check https://github.com/func-godot/func_godot_example_basic it uses `targetname` to add entities to groups for scripting purposes.

Key value pairs are just data, information, you parse it and use it however you'd like, (almost) no rules on what that data has to mean.
